src/utility/Mapping.cpp

```diff
@@ -105,4 +105,8 @@ std::vector<CoordinateAxis> get_axes(Patch patch) {
     }
     return axes;
 }
+
+std::vector<Patch> get_patches(CoordinateAxis axis) {
+    return {Patch(axis * 2), Patch(axis * 2 + 1)};
+}
```

Comment:
If you always return exactly two patches, why not return a std::tuple that way you will not have any heap allocation